### ISSUE TITLE
[BUG] Error when selecting an AAS that references a submodel not located in the Submodel Repository.

### ISSUE THREAD

**Issue by JS-Aibel** [eclipse-basyx/basyx-aas-web-ui#55]
**Describe the bug**
An error occurs when selecting an AAS that references a submodel not located in the Submodel Repository.

**To Reproduce**

In the AAS Environment do as follows.
1.	Upload an AAS with a Submodel.
2.	Delete the submodel from an API endpoint.
3.	Select the AAS in AAS UI
4.	An error occurs as shown below.
```
Error while parsing the Submodel References!
Error: Cannot read properties of undefined (reading '0')
```

**Expected behavior**
According to the specification, the AssetAdministrationShell has a ‘soft’ reference to its Submodels, meaning it does not contain Submodels. Therefor, should the viewer account for missing submodels in the submodel Repository instead of triggering an error?

**BaSyx (please complete the following information):**
- BaSyx Version [v2.0]
- BaSyx SDK [Python]
- AAS Version [DotAAS V3]

**BaSyx configuration**
Default from the Quick Start Guide.

**Comment by aaronzi** [comment]
Hi @JS-Aibel,

Thanks for reporting this. What would be the behavior you would prefer when clicking on an AAS that references a submodel that isn't there? Just show a warning?

But you are absolutely right, an uncatched error that blocks the usage of the UI is not a good behaviour.

**Comment by JS-Aibel** [comment]
Thanks for the reply, @aaronzi.

It might be helpful with a warning that explains the different types of issues:
1. **Submodel Not in Registry**: If the submodel isn’t in the registry.
2. **Submodel Fetching Issues**: If the submodel is in the registry but can’t be fetched from the repository, either because it’s missing or there’s no access to the repository.

Pop-up warnings can be easily missed or forgotten. Therefore, it would be best if the warning is visible in the viewer whenever the AAS is selected. For example, the warning could be shown as a greyed-out submodel, displaying its ID and the reason for its absence.

**Comment by aaronzi** [comment]
That's a great idea. I will add this to our backlog. We will definitely implement this!

**Comment by aaronzi** [comment]
Fixed in #73
